cob/cob.c

```diff
@@ -554,40 +666,83 @@ int m0_cob_domain_create_prepared(struct m0_cob_domain          **out,
 		.ot_type    = M0_FORMAT_TYPE_COB_DOMAIN,
 		.ot_footer_offset = offsetof(struct m0_cob_domain, cd_footer)
 	});
+	M0_BE_TX_CAPTURE_PTR(seg, tx, &dom->cd_header);
 
 	dom->cd_id = *cdid;
+	M0_BE_TX_CAPTURE_PTR(seg, tx, &dom->cd_id);
 
 	m0_format_footer_update(dom);
-	M0_BE_TX_CAPTURE_PTR(seg, tx, dom);
-
-	rc = m0_cob_domain_init(dom, seg);
-	M0_ASSERT(rc == 0); /* XXX */
-
-	M0_BE_OP_SYNC(o,
-		      m0_be_btree_create(&dom->cd_object_index, tx, &o,
-					 &M0_FID_TINIT('b',
-						       M0_BBT_COB_OBJECT_INDEX,
-						       cdid->id)));
-	M0_BE_OP_SYNC(o,
-		      m0_be_btree_create(&dom->cd_namespace, tx, &o,
-					 &M0_FID_TINIT('b',
-						       M0_BBT_COB_NAMESPACE,
-						       cdid->id)));
-	M0_BE_OP_SYNC(o,
-		      m0_be_btree_create(&dom->cd_fileattr_basic, tx, &o,
-					 &M0_FID_TINIT('b',
-						      M0_BBT_COB_FILEATTR_BASIC,
-						       cdid->id)));
-	M0_BE_OP_SYNC(o,
-		      m0_be_btree_create(&dom->cd_fileattr_omg, tx, &o,
-					 &M0_FID_TINIT('b',
-						       M0_BBT_COB_FILEATTR_OMG,
-						       cdid->id)));
-	M0_BE_OP_SYNC(o,
-		      m0_be_btree_create(&dom->cd_fileattr_ea, tx, &o,
-					 &M0_FID_TINIT('b',
-						       M0_BBT_COB_FILEATTR_EA,
-						       cdid->id)));
+	M0_BE_TX_CAPTURE_PTR(seg, tx, &dom->cd_footer);
+
+	bt = (struct m0_btree_type){.tt_id = M0_BT_COB_OBJECT_INDEX,
+		.ksize = sizeof(struct m0_cob_oikey),
+		.vsize = -1,
+	};
+	fid = M0_FID_TINIT('b', M0_BBT_COB_OBJECT_INDEX, cdid->id);
+	M0_ALLOC_PTR(dom->cd_object_index);
+	rc = M0_BTREE_OP_SYNC_WITH_RC(&b_op,
+				      m0_btree_create(&dom->cd_oi_node,
+						      sizeof dom->cd_oi_node,
+						      &bt, &b_op,
+						      dom->cd_object_index, seg,
+						      &fid, tx));
+	M0_ASSERT(rc == 0);
+
+	bt = (struct m0_btree_type){.tt_id = M0_BT_COB_NAMESPACE,
+		.ksize = -1,
```

Comment:
please check alignemnt. same comments for line no.706, 734

Reply:
done

Reply:
Resolving conversation on behalf of Radha.